packages/gatsby-remark-images-contentful/src/utils/index.js

```diff
@@ -1,20 +1,34 @@
 const axios = require(`axios`)
 
-const getBase64Img = async url => {
-  const response = await axios({
-    method: `GET`,
-    responseType: `arraybuffer`,
-    url: `${url}`,
-  })
+const getBase64Img = async (url, reporter) => {
+  let base64Img
+  try {
+    const response = await axios({
+      method: `GET`,
+      responseType: `arraybuffer`,
+      url: `${url}`,
+    })
 
-  const base64Img = `data:${
-    response.headers[`content-type`]
-  };base64,${new Buffer(response.data).toString(`base64`)}`
+    base64Img = `data:${response.headers[`content-type`]};base64,${new Buffer(
+      response.data
+    ).toString(`base64`)}`
+  } catch (err) {
+    reporter.panic(`Failed downloading the base64 image for ${url}`, err)
+  }
 
   return base64Img
 }
 
-const buildResponsiveSizes = async ({ metadata, imageUrl, options = {} }) => {
+const buildResponsiveSizes = async (
+  { metadata, imageUrl, options = {} },
+  reporter
+) => {
+  // Remove search params from the image url
+  let formattedImgUrl = imageUrl
+  if (imageUrl.includes(`?`)) {
+    formattedImgUrl = imageUrl.split(`?`)[0]
```

Comment:
Isn't this a breaking change? And potentially a bug?

In particular, this would break for image urls where the query parameter is mandatory because it's explicitly dropping the query parameter in favor of the one appended below.

An improved behavior would be to use @wardpeet's suggestion below and to add the `w` param in there as well and then to serialize that url. That way you would preserve the original query and, worst case, override the `w` query parameter.

Reply:
So far I can see its a fix, not breaking change.

If you had images in your content with query parameters before the fix, the page won't build at all. (Cus it would add a second question mark which Contentfuls Image API really does not like)